src/map_header_data.c

```diff
@@ -196,52 +196,44 @@ static void MapHeaderData_ParseEvents(MapHeaderData *data)
     const u8 *events = (const u8 *)data->tmpEventsBuf;
 
     // bgEvents
-    {
-        data->numBgEvents = *(u32 *)events;
-        events += sizeof(u32);
-        if (data->numBgEvents != 0) {
-            data->bgEvents = (const BgEvent *)events;
-        } else {
-            data->bgEvents = NULL;
-        }
-        events += sizeof(BgEvent) * data->numBgEvents;
+    data->numBgEvents = *(u32 *)events;
+    events += sizeof(u32);
+    if (data->numBgEvents != 0) {
+        data->bgEvents = (const BgEvent *)events;
+    } else {
+        data->bgEvents = NULL;
     }
+    events += sizeof(BgEvent) * data->numBgEvents;
```

Comment:
With this macro defined somewhere (probably just above this function):

```c
#define CONSUME_EVENTS(T, dataTEvents, dataNumTEvents) \
    do {                                               \
        (dataNumTEvents) = *(u32 *)events;             \
        events += sizeof(u32);                         \
        if ((numEvents) != 0) {                        \
            (dataTEvents) = (const T *)events;         \
        } else {                                       \
            (dataTEvents) = NULL;                      \
        }                                              \
        events += sizeof(T) * (dataNumTEvents);        \
    } while (0)
```

You can replace this whole block:

```suggestion
    CONSUME_EVENTS(BgEvent, data->bgEvents, data->numBgEvents);
```

And also do similarly for the other sections:

```c
    CONSUME_EVENTS(ObjectEvent, data->objectEvents, data->numObjectEvents);
    CONSUME_EVENTS(WarpEvent, data->warpEvents, data->numWarpEvents);
    CONSUME_EVENTS(CoordEvent, data->coordEvents, data->numCoordEvents);
```

Reply:
Great suggestion! Does do the do-while part matter? It seems to at least match without it as well, but I might be missing some reason for it.

Reply:
It enforces the existence of a semicolon to make it semantically resemble a function in the code which uses it. If you get rid of the semicolon with the `do while` loop still in place, then it should fail to compile the file at all, much less match.